modules/secrets/setup-secrets.nix

```diff
@@ -0,0 +1,94 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  secretsDir = "/secrets/"; # TODO: make this an option
+
+  secrets = (import ./make-secrets.nix { inherit config; }).activeSecrets;
+
+  setupSecrets = concatStrings (mapAttrsToList (n: v: ''
+    setupSecret ${n} ${v.user} ${v.group} ${v.permissions} ${optionalString (v.keyFile != null) (baseNameOf v.keyFile)}
+  '') secrets);
+in
+{
+  options.nix-bitcoin.setup-secrets = mkEnableOption "Set permissions for secrets generated by 'generate-secrets.sh'";
+
+  config = mkIf config.nix-bitcoin.setup-secrets {
+    systemd.targets.nix-bitcoin-secrets = {
+      requires = [ "setup-secrets.service" ];
+      after = [ "setup-secrets.service" ];
+    };
+
+    # Operation of this service:
+    #  - Create missing secrets that are composed of attrs from secrets.nix
+    #  - Set owner and permissions for all used secrets
+    #  - Make all other secrets accessible to root only
+    # For all steps make sure that no secrets are copied to the nix-store.
+    #
+    systemd.services.setup-secrets = {
+      serviceConfig = {
+         Type = "oneshot";
+         RemainAfterExit = true;
+      } // config.nix-bitcoin-services.defaultHardening;
+      script = ''
+        setupSecret() {
+            file="$1"
+            user="$2"
+            group="$3"
+            permissions="$4"
+            srcFile="$5"
+            if [[ ! -e $file ]]; then
```

Comment:
It seems like this would ignore if a srcFile changes. Is that what we want?

Reply:
I wanted to keep `setup-secrets.sh` as simple as possible, but you're right, we have to support that case.
This is an opportunity to simplify the secrets mechanism, in a way that is very concise to implement and maximally compatible with other deployment methods: Each deployed secret corresponds to a local secret file.
`generate-secrets.sh` would then look [like this](https://gist.github.com/7132f64c82c8b3f8b6dd01f6c4669825). To ensure backwards compatibility, we could easily convert existing old secrets to the new format in `setup.sh`. I'd strongly prefer this approach. What do you think?

Reply:
Nice idea. We don't need to fix it in this PR as setup-secrets is only used without nixops and that's not documented.

Reply:
I'd prefer to include the overhauled secrets format in this PR so we don't have to publish a bloated yet incomplete (missing change detection) `setup-secrets` implementation, only to remove it later. Would you be okay with this?

Reply:
Okay